packages/data-transport-layer/src/services/l1-ingestion/service.ts

```diff
@@ -205,7 +206,40 @@ export class L1IngestionService extends BaseService<L1IngestionServiceOptions> {
           await sleep(this.options.pollingInterval)
         }
       } catch (err) {
-        if (!this.running || this.options.dangerouslyCatchAllErrors) {
+        if (err instanceof MissingElementError) {
+          // Different functions for getting the last good element depending on the event type.
+          const handlers = {
+            SequencerBatchAppended: this.state.db.getLatestTransactionBatch,
+            StateBatchAppended: this.state.db.getLatestStateRootBatch,
+            TransactionEnqueued: this.state.db.getLatestEnqueue,
+          }
+
+          // Find the last good element and reset the highest synced L1 block to go back to the
+          // last good element. Will resync other event types too but we have no issues with
+          // syncing the same events more than once.
+          const eventName = err.name
+          const lastGoodElement: {
+            blockNumber: number
+          } = await handlers[eventName]()
```

Comment:
The safety of this depends on TS type safety that is defined in a faraway file. A future dev may break this by introducing a different `err.name` without updating the `handlers` object. Adding a check for a defined handler would be a better error message than an error about trying to invoke `undefined`